src/mono/browser/runtime/heapshot/index.ts

```diff
@@ -0,0 +1,328 @@
+/* eslint-disable no-console */
+/* eslint-disable @typescript-eslint/no-unused-vars */
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import ProductVersion from "consts:productVersion";
+import cwraps from "../cwraps";
+import { loaderHelpers } from "../globals";
+import { mono_log_info } from "../logging";
+import { getU32 } from "../memory";
+import { utf8ToString } from "../strings";
+import { BlobBuilder } from "../jiterpreter-support";
```

Comment:
I would prefer not mix jiterp and heapshot features together, because I think we may want to trim away each separately.
But let's discuss it first, maybe I don't understand well the synergy here. 

What functions make `BlobBuilder` ideal ?

Afaik that builder is allocating inside of WASM linear memory, correct ?

Reply:
blobbuilder is mostly ideal because it implements LEB and some other things we need. Duplicating that would add code size and risk, so I decided to start with reuse. It does have to use wasm linear memory, which is why I added a parachute.

Reply:
why we need LEB ?

Reply:
why memory shot needs to use linear memory for output ?